app/controllers/activities_controller.rb

```diff
@@ -1,59 +1,38 @@
 class ActivitiesController < ApplicationController
-  before_action :authorize!, only: [:show]
-  before_action :get_activity, only: [:update, :retry, :show]
+  before_action :activity, only: [:update, :retry]
 
-  # def start
-  #   activity_session.trash! unless activity_session.unstarted?
-  #   redirect_to activity_path(activity, session: activity_session.id)
-  # end
 
-  def show
-
-  end
 
   def update
     activity_session.start
     activity_session.save!
-    redirect_to activity_path(@activity, session: activity_session.id)
+    redirect_to activity_session_path(activity_session)
   end
 
   def retry
-    if current_user.nil?
-      redirect_to new_session_path
-    else
-      @activity_session = ActivitySession.new(is_retry: true, user_id: current_user.id, activity_id: @activity.id, classroom_activity_id: params[:classroom_activity_id])
-      @activity_session.start
-      @activity_session.save!
-      redirect_to activity_path(@activity, session: @activity_session.id)
-    end
+    redirect_to new_session_path if current_user.nil?
+
+    @activity_session = ActivitySession.new(is_retry: true,
+                                            user_id: current_user.id,
+                                            activity_id: @activity.id,
+                                            classroom_activity_id: params[:classroom_activity_id])
+    @activity_session.start
+    @activity_session.save!
+    redirect_to activity_session_path(@activity_session)
   end
 
-  # def resume
-  #   redirect_to activity_path(activity, session: activity_session.id)
-  # end
-
 protected
 
-  def get_activity
-    @activity = Activity.find(params[:id])
+  def activity
+    @activity ||= Activity.find(params[:id])
   end
 
   def activity_session
     @activity_session ||= if params[:anonymous]
-      :anonymous
+      nil
```

Comment:
@kriskelly do you know if #update gets called on anonymous sessions? If not, we can remove this switch...